src/core/mutatus.scala

```diff
@@ -30,25 +33,61 @@ object `package` {
     DatastoreOptions.getDefaultInstance.getService
   )
 
+  /** provides `saveAll` and `deleteAll` methods for collections of case class instances */
+  implicit class DataBatchExt[T <: Product](values: Iterable[T]) {
+
+    /** saves the all case class as a Datastore entity in batch mode */
+    def saveAll()(implicit svc: Service,
+                  encoder: Encoder[T],
+                  dao: Dao[T],
+                  idField: IdField[T]): Map[T, Ref[T]] = {
+      val (batch, refs) =
+        values.foldLeft(svc.readWrite.newBatch() -> mutable.Map.empty[T, Ref[T]]) {
+          case ((b, entityRefs), value) =>
+            val entity = value.buildEntity()
+            b.put(value.buildEntity())
+            b -> entityRefs.updated(value, new Ref[T](entity.getKey))
+        }
+      batch.submit()
+      refs.toMap //batch may supply only those keys which were generated by Datastore, but in mutatus generates keys deterministically based on entity content
+    }
+
+    /** deletes the Datastore entities in batch mode*/
+    def deleteAll()(implicit svc: Service, dao: Dao[T], idField: IdField[T]): Unit = {
+      val _ = values
```

Comment:
What's the point in `val _ =`?

Reply:
As method returns Unit and last call returns some value there may be warning at compile about "discarded non-unit value". I'm not sure is it important when using Fury, however it was done because of my habit and experience with IntelliJ.

Reply:
Ah, I usually use an extension method (`.unit`) for this purpose. I call on the last item, and it converts it to a `Unit`. There's no point adding the extension method just for that, so let's just have the warning for now.